src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorFeatureDetector.cs

```diff
@@ -34,6 +36,7 @@ public DefaultLSPEditorFeatureDetector(ProjectHierarchyInspector projectHierarch
             }
 
             _projectHierarchyInspector = projectHierarchyInspector;
+            _featureFlags = (IVsFeatureFlags)AsyncPackage.GetGlobalService(typeof(SVsFeatureFlags));
```

Comment:
❗ MEF constructors _must not_ call GetGlobalService (or any variation)

Reply:
> ❗ MEF constructors _must not_ call GetGlobalService (or any variation)

Ah thanks for heads up Sam! We had this there for many releases before but I can move it down into our lazily evaluated "use editor" invocation. Out of curiosity, what's the restriction you're referring to?

Reply:
https://github.com/dotnet/roslyn/issues/15148